pkg/viaduct/pkg/conn/ws.go

```diff
@@ -151,6 +151,11 @@ func (conn *WSConnection) handleMessage() {
 
 func (conn *WSConnection) SetReadDeadline(t time.Time) error {
 	conn.ReadDeadline = t
+	conn.locker.Lock()
```

Comment:
It seems that no locks need to be added here, the wsConn field is created at initialization, and these methods also seem to have no concurrency issues.